lib/middleware/routes-metadata/routes-metadata.js

```diff
@@ -237,26 +339,69 @@ const pageHandler = async (req, res, next) => {
       await userData.saveData()
     }
 
+    // update visited records
```

Comment:
Moved this so that it only gets run if we are actually going to render the page.

Added params to visited pages bundle, and using params in addition to _id to check whether to add page to list.

Also added explicit check that it's a page that collects user data for the actual form (ie. scope 'input' rather than'savereturn')